src/utils/jump_utils.jl

```diff
@@ -1,6 +1,6 @@
 #Given the changes in syntax in ParameterJuMP and the new format to create anonymous parameters
-function add_jump_parameter(model::JuMP.Model, val::Number)
-    param = JuMP.@variable(model, variable_type = PJ.Param())
+function add_jump_parameter(jump_model::JuMP.Model, val::Number)
```

Comment:
@odow if you have a chance can you take a look at this file, please? Just to make sure I am not violating any JuMP usage principle.

Reply:
Seems okay. 

The main principle is to avoid optimize-modify-query. So 
```Julia
for x in all_variables(model)
    v = value(x)
    fix(x, v; force = true)
end
```
is bad because it is query-modify-query-modify.

Do instead:
```Julia
solution = Dict(x => value(x) for x in all_variables(model))
for (x, v) in solution
    fix(x, v; force = true)
end
```
The problem is that this could be quite hard to notice if the query and modifications are spread across different functions or files.

Reply:
So this thing that we do to warm start: 

```
function _apply_warm_start!(model::DecisionModel)
    container = get_optimization_container(model)
    jump_model = get_jump_model(container)
    all_vars = JuMP.all_variables(jump_model)
    all_vars_value = JuMP.value.(all_vars)
    JuMP.set_start_value.(all_vars, all_vars_value)
    return
end
```

Reply:
Yes. If after doing this you then went and queried something else.

For most solvers you're probably okay, but there might be some solvers that break. (I think SCS would, in this case. Setting the primal start would cause the caching optimizer to reset, invalidating the solution.)

Reply:
I will add a note to make sure we query before doing this.

Reply:
Addressed in fcca97a